### DISCUSSION TITLE
[Round: 0x02]在starcoin console中调用MyCounter的方法为什么可以随意调用而无需签名

### DISCUSSION THREAD

**Discussion by hi-chenerge** [NonceGeek/Move-Camp#223] [Q&A] [answered]
account execute-function --function 0x3da0f104118cca58ae1ad4272ae59008::MyCounter::incr_counter -s 0x3da0f104118cca58ae1ad4272ae59008 -b

-s 是account地址，这里为啥不签名，那谁都可以随意调用了，是因为是在console中就不限制了嘛

**Comment by v1xingyue** [comment, marked as answer]
调用需要签名的函数的时候，需要提前使用unlock 解锁当前账户。 然后，就可以使用当前账户的私钥了。 
console 客户端 程序，自动完成了签名操作。

**Comment by mason0510** [comment]
老铁 默认时间内 输过密码 你的钱包已经解锁了

**Comment by Demian101** [comment]
后面在 aptos 和 sui 的案例中，可以看到类似函数需要传入 signer ， 这个 signer 就会要求你的签名操作。
